app/middleware/otp-verify.js

```diff
@@ -1,59 +0,0 @@
-'use strict'
-
-const paths = require('../paths')
-const logger = require('../utils/logger')(__filename)
-const errorResponse = require('../utils/response').renderErrorView
-const registrationService = require('../services/service-registration.service')
-const { validateOtp } = require('../utils/registration-validations')
-
-// Exports
-module.exports = {
-  verifyOtpForServiceInvite
-}
-
-// Middleware methods
-
-/**
- * Process submission of otp verification
- *
- * @param req
- * @param res
- * @param next
- * @returns {*|Promise|Promise.<T>}
- */
-function verifyOtpForServiceInvite (req, res, next) {
-  const correlationId = req.correlationId
-  const code = req.register_invite.code
-  const otpCode = req.body['verify-code']
-
-  const handleInvalidOtp = (message) => {
-    logger.debug(`[requestId=${correlationId}] invalid user input - otp code`)
-    req.flash('genericError', message)
-    res.redirect(303, paths.selfCreateService.otpVerify)
-  }
-
-  return validateOtp(otpCode)
-    .then(() => {
-      return registrationService.submitServiceInviteOtpCode(code, otpCode, correlationId)
-    })
-    .then(next)
-    .catch(err => {
-      logger.warn(`[requestId=${req.correlationId}] Invalid invite code attempted ${req.code}, error = ${err.errorCode}`)
-      switch (err.errorCode) {
-        case undefined:
-          handleInvalidOtp(err.message)
-          break
-        case 401:
-          handleInvalidOtp('Invalid verification code')
-          break
-        case 404:
-          errorResponse(req, res, 'Unable to process registration at this time', 404)
```

Comment:
As discussed, adding assertions to the existing tests (where possible) on this branching logic would be good. Otherwise everything looks great